src/main/java/org/brapi/test/BrAPITestServer/factory/geno/SampleComponent.java

```diff
@@ -0,0 +1,54 @@
+package org.brapi.test.BrAPITestServer.factory.geno;
+
+import io.swagger.model.Metadata;
+import io.swagger.model.core.BatchTypes;
+import io.swagger.model.geno.Sample;
+import io.swagger.model.geno.SampleSearchRequest;
+import org.brapi.test.BrAPITestServer.factory.BrAPIComponent;
+import org.brapi.test.BrAPITestServer.model.entity.SearchRequestEntity;
+import org.brapi.test.BrAPITestServer.service.geno.SampleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class SampleComponent implements BrAPIComponent<Sample, SampleSearchRequest> {
+    private final SampleService sampleService;
+
+    @Autowired
+    public SampleComponent(SampleService sampleService) {
+        this.sampleService = sampleService;
+    }
+
+    @Override
+    public List<Sample> findEntities(SampleSearchRequest request, Metadata metadata) {
+        return sampleService.findSamples(request, metadata);
+    }
+
+    @Override
+    public BatchTypes getBatchType() {
+        return BatchTypes.SAMPLES;
+    }
+
+    @Override
+    public SearchRequestEntity.SearchRequestTypes getSearchType() {
```

Comment:
This method is never used. please remove (see comment in BrAPIComponent.java

Reply:
done

Reply:
Marking resolved - this was removed